source/_posts/2017-03-30-Haptic-feedback-makes-you-vibrate.md

```diff
@@ -0,0 +1,147 @@
+
+The impact type of feedback is to be used to enhance a visual experience. For example, it can be used in a game when a collision occurs. I couldn't find this documented, but I'm almost sure that Impact - heavy is the type of feedback that the user feels when a peeked view pops full screen.
+
+**Light**
+
+<video width="400" controls>
+  <source src="http://nodesmanager.io/haptic/impact_light.mp4" type="video/mp4">
+  Your browser does not support HTML5 video.
+</video>
+
+
+```swift
+let lightImpactFeedbackGenerator = UIImpactFeedbackGenerator(style: .light)
+lightImpactFeedbackGenerator.impactOccurred()
+```
+
+**Medium**
+
+<video width="400" controls>
+  <source src="http://nodesmanager.io/haptic/impact_medium.mp4" type="video/mp4">
+  Your browser does not support HTML5 video.
+</video>
+
+```swift
+let mediumImpactFeedbackGenerator = UIImpactFeedbackGenerator(style: .medium)
+mediumImpactFeedbackGenerator.impactOccurred()
+```
+
+**Heavy**
+
+<video width="400" controls>
+  <source src="http://nodesmanager.io/haptic/impact_heavy.mp4" type="video/mp4">
+  Your browser does not support HTML5 video.
+</video>
+
+```swift
+let heavyImpactFeedbackGenerator = UIImpactFeedbackGenerator(style: .heavy)
+heavyImpactFeedbackGenerator.impactOccurred()
+
+```
+
+### Notification
+
+The notification type of feedback should be used to indicate to the user that a task has completed. Depending on the outcome of the task (success, error, warning), the feedback pattern differs.
+
+**Success**
+
+<video width="400" controls>
+  <source src="http://nodesmanager.io/haptic/success.mp4" type="video/mp4">
+  Your browser does not support HTML5 video.
+</video>
+
+```swift
+let successNotificationFeedbackGenerator = UINotificationFeedbackGenerator()
+successNotificationFeedbackGenerator.notificationOccurred(.success)
+```
+
+**Warning**
+
+<video width="400" controls>
+  <source src="http://nodesmanager.io/haptic/warning.mp4" type="video/mp4">
+  Your browser does not support HTML5 video.
+</video>
+
+```swift
+let warningNotificationFeedbackGenerator = UINotificationFeedbackGenerator()
+warningNotificationFeedbackGenerator.notificationOccurred(.warning)
+```
+
+**Failure**
+
+<video width="400" controls>
+  <source src="http://nodesmanager.io/haptic/error.mp4" type="video/mp4">
+  Your browser does not support HTML5 video.
+</video>
+
+```swift
+let errorNotificationFeedbackGenerator = UINotificationFeedbackGenerator()
+errorNotificationFeedbackGenerator.notificationOccurred(.error)
+```
+
+
+Using the `UIFeedbackGenerator` may have a little latency. If you want to use it to match some sound or visual effects, you can call the `prepare()` method on the generator to put the Taptic Engine in a prepared state and reduce the latency. It will stau prepared a few seconds, or until the next feedback is triggered.
+
+
+## When to use haptic feedback
+
+We saw that creating haptic feedback is super easy. However, the complicated part is to know when and where to use it.
```

Comment:
to know -> knowing 

(this is the most minor thing ever so you can ignore it if you want 😃 )